### DISCUSSION TITLE
Installation on Mac doesn't work

### DISCUSSION THREAD

**Discussion by patrickcneuhaus** [diesel-rs/diesel#2836] [Q&A]
Hey guys,

Running `cargo install diesel_cli --no-default-features --features postgres` on a Mac (Big Sur 11.2.3) leads to the following error:

```
 note: ld: library not found for -lpq
          clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

I already checked out dozens of discussions about this error, and tried multiple fixes (e.g. `brew install postgres`, `brew install libpq`), but nothing worked.

Does someone have a solution for this?

**Comment by weiznich** [comment]
This error message indicates that the compiler cannot find a compatible version of `libpq`. That's an issue with your environment, so you need to fix it there. As I don't own a Mac I can only point you to our [CI configuration](https://github.com/diesel-rs/diesel/blob/master/.github/workflows/ci.yml#L105-L112), as things work there.

**Comment by heru** [comment]
> Hey guys,
> 
> Running `cargo install diesel_cli --no-default-features --features postgres` on a Mac (Big Sur 11.2.3) leads to the following error:
> 
> ```
>  note: ld: library not found for -lpq
>           clang: error: linker command failed with exit code 1 (use -v to see invocation)
> ```
> 
> I already checked out dozens of discussions about this error, and tried multiple fixes (e.g. `brew install postgres`, `brew install libpq`), but nothing worked.
> 
> Does someone have a solution for this?
If you have homebrew, just run brew install diesel.
1. install homebrew
2. install diesel cli from homebrew: brew install diesel.

if you don't want bloated your mac with all features, just install necessary lib. for example if you just want only postgresql the steps are:
1. install libpq: brew install libpq.
2. follow instruction after libpq successfully installed. for example here at my mac
     
<img width="624" alt="image" src="https://user-images.githubusercontent.com/28038/127743504-b4b6d192-4eb5-483d-b925-04e20d7d372a.png">
Just follow instruction above

3. reload your .zshrc or .bashrc

4. install diesel cli from cargo: cargo install diesel_cli --no-default-features --features postgres
